Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -109,4 +119,40 @@ private extension PointOfSaleCouponService {
             }
         }
     }
+
+    private func checkStoreCouponSettings() async -> Bool {
+        await withCheckedContinuation { continuation in
+            let action = SettingAction.retrieveCouponSetting(siteID: siteID) { result in
+                switch result {
+                case let .success(isEnabled):
+                    debugPrint("Coupons enabled? \(isEnabled)")
+                    continuation.resume(returning: isEnabled)
+                case let .failure(error):
+                    debugPrint("Coupons settings error: \(error)")
+                    continuation.resume(returning: false)
+                }
+            }
+            Task { @MainActor in
+                stores?.dispatch(action)
+            }
+        }
+    }
+
+    private func enableStoreCouponSettings() async -> Bool {
```

Comment:
This is not used at the moment, happy to remove it or keep it if you prefer that we handle the enable action via CTA on this same PR 👍

Reply:
If we're not handling enabling on this PR, then let's remove it. 👍

Reply:
Done! https://github.com/woocommerce/woocommerce-ios/pull/15468/commits/d2619deecaf3ebd4171161ea476a1bd9a89c96e9